src/crimson/os/seastore/device.cc

```diff
@@ -31,13 +34,24 @@ std::ostream& operator<<(std::ostream& out, const device_config_t& conf)
 }
 
 seastar::future<DeviceRef>
-Device::make_device(const std::string& device)
+Device::make_device(const std::string& device, device_type_t dtype)
 {
-  // TODO: configure device type
-  return SegmentManager::get_segment_manager(device
-  ).then([](DeviceRef ret) {
-    return ret;
-  });
+  if (get_default_backend_of_device(dtype) == backend_type_t::SEGMENTED) {
+    return SegmentManager::get_segment_manager(device
+    ).then([](DeviceRef ret) {
+      return ret;
+    });
+  } else {
+    auto rbm = std::make_unique<NVMeBlockDevice>();
```

Comment:
This part is not used in current code, right? I think it is a better way to remove this change (else {}) for readability. I will add rbm initialization step, probably this part can be added after then. Otherwise, looks good to me.

Reply:
ok, remove if condition and add `assert(get_default_backend_of_device(dtype) == backend_type_t::SEGMENTED)`.